### DISCUSSION TITLE
How to use it with blade?

### DISCUSSION THREAD

**Discussion by maxi032** [aerni/statamic-advanced-seo#103] [Ideas]
Nice addon. It seems the most complete solution so far but my problem is that we are using blade and I don't know how to fetch the data.
I've tried 
`Statamic::tag('seo:title')->fetch()` to get the site title.
Also, in a template used by a blueprint I would need to have Entry meta_title | Site title

**Comment by aerni** [comment]
You have to pass the context to the tag to make it work:

```blade
{{ Statamic::tag('seo:title')->context($__data) }}
```

If you encounter other issues, please post the actual error you are running into.

**Comment by maxi032** [comment]
Thanks a lot @aerni for your reply but how do I access properties from context? Sorry but I am new to Statamic. My data is inside pageData:
![pageData2](https://user-images.githubusercontent.com/1525885/201349041-4f577b33-d5f3-4337-8f82-75199c41df4c.jpg)
If I do
```
{{ Statamic::tag('seo:title')->context($__data) }}
``` 
I get nothing inside <title></title> tags. 
Same with 
```
{{ Statamic::tag('seo:seo_title')->context($__data) }}
```

**Comment by aerni** [comment]
I don't understand what you're trying to do. Do you want to get data from an entry or use the Advanced SEO tags? That's a difference.

If you want to output the head tag with all the data like title, meta description, social images, etc. You should use this in your layout view:

```
<head>{{ Statamic::tag('seo:head')->context($__data) }}</head>
```

If you simply want to get data from the entry, you do it like this:

```
{{ $page->seo_title }}
{{ $page->seo_description }}
...
```

**Comment by stuartcusackie** [comment]
This is a little complicated if you are using Layout Components. The SEO Pro addon solved this by automatically pulling the context when it is missing, see [here](https://github.com/statamic/seo-pro/pull/235/files#diff-e2da0460fea071b3322df1134dc0257058058752ec14b97886b2f961548fb04e).

It would be nice if this addon did the same.

**Comment by aerni** [comment]
I don't use Blade myself. Feel free to create a sample repo that recreates how you'd like to use the addon. I'll gladly take a look to implementing this.

**Comment by jonathan-bird** [comment]
@aerni I use blade exclusively, and using this addon. Very happy with how it all works so far, just would be great to not have to pass context (or at minimum, please add to your docs as Blade is more used in Statamic 4)

**Comment by aerni** [comment]
Noted. I'll look into this when I get to it.

**Comment by aerni** [comment]
Better Blade support is coming. See https://github.com/aerni/statamic-advanced-seo/pull/152.
